autodynatrace/wrappers/custom/wrapper.py

```diff
@@ -5,23 +5,55 @@
 from ...sdk import sdk
 
 
+def use_fully_qualified_name():
+    return os.environ.get("AUTODYNATRACE_CUSTOM_SERVICE_USE_FQN") \
+           and os.environ.get("AUTODYNATRACE_CUSTOM_SERVICE_USE_FQN").lower() == 'true'
+
+
+def get_custom_defined_service_name():
+    return os.environ.get("AUTODYNATRACE_CUSTOM_SERVICE_NAME")
+
+
 def generate_service_name(wrapped):
-    if os.environ.get("AUTODYNATRACE_CUSTOM_SERVICE_NAME"):
-        return os.environ.get("AUTODYNATRACE_CUSTOM_SERVICE_NAME")
+    if get_custom_defined_service_name():
+        return get_custom_defined_service_name()
+    else:
+        return get_module_path(wrapped)
+
+
+def get_module_path(wrapped):
+    module_path = wrapped.__module__
+    class_name = None
+    qual_name = None
+    result = module_path
 
-    service_name = wrapped.__module__
     if hasattr(wrapped, "im_class"):
-        service_name = wrapped.im_class.__name__
+        class_name = wrapped.im_class.__name__
+        result = class_name
+
     if hasattr(wrapped, "__qualname__") and "." in wrapped.__qualname__:
-        service_name = wrapped.__qualname__.split(".")[0]
+        qual_name = wrapped.__qualname__.split(".")[0]
+        result = qual_name
+
+    if use_fully_qualified_name():
+        result = ".".join([i for i in [module_path, class_name, qual_name] if i])
+
+    return result
+
 
-    return service_name
+def generate_method_name(wrapped):
+    name = wrapped.__name__
+    if get_custom_defined_service_name() or use_fully_qualified_name():
+        path = get_module_path(wrapped)
+        return path + "." + name
```

Comment:
Can we use `return "{}.{}".format(path, name)` here to keep the style of the other string manipulations?